src/spatial/utils.jl

```diff
@@ -0,0 +1,219 @@
+"""
+A file with types, structs and functions for spatial simulations
+"""
+
+"""
+stores info for a spatial jump
+"""
+struct SpatialJump{J}
+    site::J #where the jump happens
+    index::Int #diffusion or reaction
+    target_site::J #target of diffusive hop
+end
+
+############ abstract spatial system struct ##################
+"""
+Contains all info about the topology of the system
+"""
+abstract type AbstractSpatialSystem end
+
+"""
+returns neighbors of site
+"""
+function neighbors end
```

Comment:
What advantage is there to actually declaring these (empty) functions vs. just leaving a comment that this is the API? I think it would perhaps be better to just comment them out here but make clear they are the API / public interface in documentation. (Happy to hear arguments against that though.)

Reply:
One advantage is that (as far as I understand) the docstring will apply to all methods of this abstract function. Another is that perhaps some of them can be implemented from each other, such that `get_site_rate` for `AbstractSpatialRates`. In OOP languages this is standard practice. I am happy to change it if you say so.